### PR TITLE
Fix some numpy deprecation warnings

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

As I was searching logs for an explicit mention of a future version of numpy that will break the current state of yt, I was able to fix _some_ of the existing warnings (definitely not all, some are much harder to track down than others).

### CODE REVIEW

**Review by cphyc — APPROVED** [review]

File: yt/data_objects/tests/test_points.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def test_fast_find_field_values_at_points():
     # sample 100 particles
     nparticles = 100
     ppos = ad["all", "particle_position"]
-    ppos = ppos[np.random.random_integers(0, len(ppos) - 1, size=nparticles)]
+    ppos = ppos[np.random.randint(low=0, high=len(ppos), size=nparticles)]
```
Comment (matthewturk) [inline yt/data_objects/tests/test_points.py:65]: Just verifying: `high` is not inclusive, right?

File: yt/testing.py [hunk 2]
```diff
@@ -729,8 +729,7 @@ def add_noise_fields(ds):
 
     def _binary_noise(field, data):
         """random binary data"""
-        res = prng.random_integers(0, 1, data.size).astype("float64")
-        return res
+        return prng.randint(low=0, high=2, size=data.size).astype("float64")
```
Comment (matthewturk) [inline yt/testing.py:732]: I see from *this* one that it isn't!  Never mind the above.

**Review by matthewturk — APPROVED** [review]
lgtm

File: yt/data_objects/tests/test_points.py (hunk 1 above)
Comment (neutrinoceros) [reply yt/data_objects/tests/test_points.py:65]: yes, that's the important differences between the two apis, and it's advertised in the deprecation warning itself :)

**Pull request merged**